src/Facility.Core/ServiceErrorDto.g.cs

```diff
@@ -35,7 +35,7 @@ public ServiceErrorDto()
 		/// <summary>
 		/// Advanced error details.
 		/// </summary>
-		public JObject? Details { get; set; }
+		public System.Collections.Generic.Dictionary<string, object?>? Details { get; set; }
```

Comment:
Right, I was kinda just testing out how to make it work without a serializer-specific type. Something like this might work for maintaining compatibility:
```csharp
[Obsolete]
[JsonIgnore]
public JObject? Details
{
    get => Map(DetailInfo);
    set => DetailInfo = Map(value);
}

[JsonPropertyName("details")]
public Dictionary<string, object>? Details { get; set; }
```

Reply:
Thoughts on that approach?

Reply:
The approach for maintaining backward compatibility with the old `Details` property seems good, but I don't love the ambiguity of the `object` dictionary value type.

I've also pondered introducing a `ServiceObject` class (akin to `ServiceResult`) that represents an arbitrary JSON object but supports multiple representations, e.g. via `JObject` and `JsonObject` properties. It would be used by the new code generator everywhere `object` is used in the FSD.

```csharp
[Obsolete("Use DetailsObject.)]
[JsonIgnore]
public JObject? Details
{
    get => DetailsObject?.JObject;
    set => DetailsObject = value is null ? null : new ServiceObject(value);
}

[JsonPropertyName("details")]
public ServiceObject? DetailsObject { get; set; }
```

Reply:
I don't love the ambiguity of `object` either, but I do love the simplicity. The upsides are it's easy to use and works with any serializer. The downside is it's easy to use a non-serializable type, and that's less likely to be caught in testing because it's an error condition.

`ServiceObject` with a different property for each serializer would be slightly annoying, and consumers would have to ensure the property accessed agreed with the serializer configured.

A third option would be our own JSON node types.
```c#
abstract class ServiceNode {}
class ServiceObject : ServiceNode, IDictionary<string, ServiceNode> {}
class ServiceArray : ServiceNode, ICollection<ServiceNode> {}
class ServiceValue : ServiceNode
{
    string ToString();
    int ToInt32();
    double ToDouble();
    bool ToBoolean();
}
```
That maybe good enough for error details, but probably not anything else.

Reply:
> consumers would have to ensure the property accessed agreed with the serializer configured

I figured we could convert one from the other as necessary, but I suppose mutability makes that a bit complex. We could standardize on `System.Text.Json` (i.e. it uses a property, and there are `FromJObject` and `ToJObject` methods).

The object-valued dictionary is also problematic in that regard, since the serializer determines what types are serializable/deserialized.

> A third option would be our own JSON node types.

This is tempting but feels too complex. I'd rather standardize on the `System.Text.Json` API and make interop easy (if less optimized) for `Newtonsoft.Json` clients. To me, this seems simple, predictable, and has no unknowns that require documentation to resolve.

Reply:
I was thinking the dependency on .NET 6 would be problematic, but System.Text.Json is available as a package with a net461 target. I'm on board with using the thing that's first-class, built in.